### PR TITLE
New AutoStartBehavior enum for features

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
Overview
---
New AutoStartBehavior enum for features

Changes
---
- Adds new "AutoStartBehavior" enum to the Core Utilities for adoption by features.

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Definitions/Utilities/AutoStartBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    /// The first causes the feature to start immediately. The second allows the feature to be manually started at a later time.
+    /// </summary>
+    public enum AutoStartBehavior
+    {
+        /// <summary>
+        /// Automatically start the feature
+        /// </summary>
+        AutoStart = 0,
+        /// <summary>
+        /// Delay the start of the feature until the user requests recognition to begin
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Definitions/Utilities/AutoStartBehavior.cs]: "recognition"? Maybe "it"?

**Comment by davidkline-ms** [comment]
Not to be a pain on this, but what task does this address? Is this based on the discussion from today's shiproom meeting (re: ability to start systems suspended)?

File: Assets/MixedRealityToolkit/_Core/Definitions/Utilities/AutoStartBehavior.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit/_Core/Definitions/Utilities/AutoStartBehavior.cs]: feck

**Comment by StephenHodgson** [comment]
I asked Simon to pull it out as a separate PR so I can use it for gesture recognition.

**Comment by SimonDarksideJ** [comment]
This needs to be merged in BEFORE #2691

**Comment by StephenHodgson** [comment]
It's also blocking #2705

**Comment by StephenHodgson** [comment]
>  (re: ability to start systems suspended)?

No this is only for recognizers like gesture, speech, dictation, etc.

**Comment by StephenHodgson** [comment]
It could even be used for more than just that. It's for anything really.

File: Assets/MixedRealityToolkit/_Core/Utilities/AutoStartBehavior.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/_Core/Utilities/AutoStartBehavior.cs]: The namespace doesn't match the folder. Shouldn't this be in the `Assets/MixedRealityToolkit/_Core/Definitions/Utilities` folder?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Utilities/AutoStartBehavior.cs]: I didn't realize he moved it 😅

**Pull request merged**